support form: add error messages to file section

jennur opened the pull request:
Closes https://github.com/zenodo/zenodo-rdm/issues/81

- Adds error message from `Dropzone` to the UI
- I decided to add the `Message` component from SUI instead of raw text for the error, as I find it more visually pleasing. Let me know if it should not be used here.
- The "max size" error shows when the *total* size exceeds the defined size limit, and clears when the user uploads new files.
    - When uploading multiple files, after the size limit has been reached, no more files are uploaded (as seen in the last screenshot, where 8 files were attempted uploaded, but only 6 made it through).
- I was not able to test the wrong format error, as it is not possible to upload the wrong format in the file-select dialog. 

## Screenshots
<img width="1186" alt="Screenshot 2022-11-17 at 15 04 07" src="https://user-images.githubusercontent.com/21052053/202468953-ceb82378-b2e0-4c6c-ac7d-e574a36707ea.png">
<img width="1183" alt="Screenshot 2022-11-17 at 15 03 37" src="https://user-images.githubusercontent.com/21052053/202468973-3d798868-0a4a-4e9f-846d-474ae918a339.png">

kpsherva commented on site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js:
```diff
@@ -41,6 +41,11 @@ class SupportForm extends Component {
         super(props);
+
+        this.state = {
+            fileErrorMessage: "",
```
```suggestion
            fileErrorMessage: undefined,
```

kpsherva commented on site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js:
```diff
@@ -120,12 +125,32 @@ class SupportForm extends Component {
+                        },
+                        onDrop: () => {
+                            this.setState({ 
+                                fileErrorMessage: "",
```
undefined

kpsherva commented on site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js:
```diff
@@ -120,12 +125,32 @@ class SupportForm extends Component {
+                            if(totalFileSize + file.size > maxFileSize) {
+                                return {
+                                    code: "file-too-large",
+                                    message: ""
```
I guess we don't need to return the message since is empty

kpsherva commented on site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js:
```diff
@@ -120,12 +125,32 @@ class SupportForm extends Component {
-                            console.error(rejectedFiles[0].errors);
+                            const error = rejectedFiles[0].errors[0];
+                            let message = error.message;
+                            if(error.code === "file-too-large") {
+                                message = "Max attachment size exceeded. Please add URLs to the files or make a smaller selection.";
+                            }
```
I think it would be better to have this separately defined in an object:
```
ErrorCodes = { 
"file-too-large": "Max attachent ...."
}
```
and then you simply use as
```
this.setState({ fileErrorMessage: ErrorCodes[code] });
```

jennur replied:
Thanks!!

kpsherva commented on site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js:
```diff
@@ -120,12 +125,32 @@ class SupportForm extends Component {
-                            // TODO: show error message when files are rejected e.g size limit
-                            console.error(rejectedFiles[0].errors);
+                            const error = rejectedFiles[0].errors[0];
+                            let message = error.message;
```
this assignment is not needed

alejandromumo commented:
We realized we have two PRs for this (https://github.com/zenodo/zenodo-rdm/pull/89). 

We are closing the other PR and keeping this one.

I will copy/paste the discussion points from the other PR here.

alejandromumo commented:
@zzacharo [comment](https://github.com/zenodo/zenodo-rdm/pull/89#discussion_r1022763072):

```
I think for consistency with the other instances, I would put the error summary at the top of the form...
```

alejandromumo commented:
@jrcastro2 [comment](https://github.com/zenodo/zenodo-rdm/pull/89#discussion_r1023935305) on `file-too-large`:

> Comment: IMHO I would try to avoid this, I think the backend should return a human readable error message.

alejandromumo commented on site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/FileUploader.js:
```diff
@@ -19,16 +21,24 @@ const FileUploader = ({ dropzoneParams, maxFileSize, name, currentFiles, handleD
                         }
+                        
+                        {errorMessage &&
+                            <Message negative>
```
@zzacharo [comment](https://github.com/zenodo/zenodo-rdm/pull/89#discussion_r1022763072):

> I think for consistency with the other instances, I would put the error summary at the top of the form...

jennur replied:
I think there is a misunderstanding here, as we don't have an error summary for all the fields, only highlighted field-errors.

alejandromumo commented on site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js:
```diff
@@ -120,12 +125,32 @@ class SupportForm extends Component {
-                            console.error(rejectedFiles[0].errors);
+                            const error = rejectedFiles[0].errors[0];
+                            let message = error.message;
+                            if(error.code === "file-too-large") {
```
@jrcastro2 [comment](https://github.com/zenodo/zenodo-rdm/pull/89#discussion_r1023935305) on `file-too-large`:

> Comment: IMHO I would try to avoid this, I think the backend should return a human readable error message.

jennur commented:
## Updated screenshots
![Screenshot 2022-11-18 at 11 32 16](https://user-images.githubusercontent.com/21052053/202683024-672692a4-8413-437a-a7c3-b6c2e2ba5bca.png)
![Screenshot 2022-11-18 at 11 32 03](https://user-images.githubusercontent.com/21052053/202683119-cccdde32-b805-47d9-b287-269aa9770291.png)

jrcastro2 commented on site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/FileTable.js:
```diff
@@ -8,9 +8,9 @@ import {
+const FileTable = ({ filesList, handleDelete, decimalSizeDisplay, negative }) => {
     return (
-        <Table>
+        <Table className={negative ? "negative" : ""}>
```
Minor: I think that the [Table cmp](https://react.semantic-ui.com/collections/table/#states-positive-negative) from SUIR has a negative prop that could be used in this case and avoid the custom css, WDYT?
```suggestion
        <Table>
```
Down you could use:
```diff
- <Table.Row key={file.name}>
+ <Table.Row key={file.name} negative={negative}>
```

jennur replied:
Oh I didn't know! Will check it out, thanks!

kpsherva replied:
werent we using the class negative instead the props because of UX related color theme overrides ? or those were different components, not table?

jennur commented:
## Updated screenshot
- Removed `.negative` class from `table.overrides`, and added SUI's `negative`-prop to `<Table.Row />` as suggested by @jrcastro2.
![Screenshot 2022-11-24 at 10 58 33](https://user-images.githubusercontent.com/21052053/203754741-44722475-a9f5-4d44-946c-1a32d88de434.png)

kpsherva reviewed (commented):
LGTM! just a small doubt above about the negative prop
